src/parser/mod.rs

```diff
@@ -12486,7 +12486,7 @@ impl<'a> Parser<'a> {
                 };
                 let mut relation = self.parse_table_factor()?;
 
-                if self.peek_parens_less_nested_join() {
+                if dialect_of!(self is SnowflakeDialect) && self.peek_parens_less_nested_join() {
```

Comment:
Can we change this to a dialect method that is only set to true for snowflake? e.g. `if self.dialect.supports_parens_less_nested_join() && self.peek_parens_less_nested_join()`